design/nfs/nfs-quota.md

```diff
@@ -0,0 +1,68 @@
+# NFS Quota
+
+## Background
+
+Currently, when the user creates PersistentVolumes from an NFS Rook share/export via PersistentVolumeClaim, the provisioner does not provide the spesific capacity as requested. For example the users create PersistentVolumeClaim as following:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: rook-nfs-pv-claim
+spec:
+  storageClassName: "rook-nfs-share"
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Mi
+```
+
+The client still can use the higher capacity than `1mi` as requested.
+
+This proposal is to add features which the Rook NFS Provisioner will provide the specific capacity as requested from `.spec.resources.requests.storage` field in PersistentVolumeClaim.
+
+## Implementation
+
+The implementation will be use `Project Quota` on xfs filesystem. When the users need to use the quota feature they should use xfs filesystem with `prjquota/pquota` mount options for underlying volume. Users can specify filesystem type and mount options through StorageClass that will be used for underlying volume. For example:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard-xfs
+parameters:
+  fsType: xfs
+mountOptions:
+  - prjquota
+...
+```
+
+Then the underlying PersistentVolumeClaim should be used that StorageClass 
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: nfs-default-claim
+spec:
+  storageClassName: "standard-xfs"
+  accessModes:
+  - ReadWriteOnce
+...
+```
+
+If the above conditions are met then the Rook NFS Provisioner will create projects and set the quota limit using [xfs_quota](https://linux.die.net/man/8/xfs_quota) before creating PersistentVolumes based on `.spec.resources.requests.storage` field in PersistentVolumeClaim. Otherwise the Rook NFS Provisioner will provision a PersistentVolumes without creating setting the quota.
```

Comment:
what does it look like to "create projects and set the quota limit", can you add an example command that we will call?

do we need to update the container image to now also include this `xfs_quota` binary?

Reply:
yeah, I'll add the example command for creating projects and setting the quota limit that will be invoked by nfs-provisioner.

yes, we need to update the container image to include the `xfs_quota` binary. will update it on the doc as well